### ISSUE TITLE
Getting an error when using place-order

### ISSUE THREAD

**Issue by grossman8** [blockworks-foundation/mango-explorer#37]
Any ideas why I'm getting this error?  I have a funded account on mango with about $10 in it.  

```
./place-order --market SOL/USDC --quantity 0.01 --price 90.00 --side BUY --order-type LIMIT --account-address 2WkWHY1u7LyXuYZSjA2SEqNiZDyhr8VMrNQ8rfVtBxfk
2022-02-01 18:51:13 ⚠ root         

2022-02-01 18:51:14 ⓘ SpotMarketOp Placing SOL/USDC order « Order BUY for 0.01000000 at 90.00000000 [ID: 0 / 0] LIMIT».
Traceback (most recent call last):
  File "/root/mango/mango-explorer/bin/./place-order", line 41, in <module>
    placed = market_operations.place_order(order)
  File "/root/mango/mango-explorer/mango/spotmarketoperations.py", line 188, in place_order
    transaction_ids = (signers + place + crank + settle).execute(self.context)
  File "/root/mango/mango-explorer/mango/combinableinstructions.py", line 242, in execute
    raise exception
  File "/root/mango/mango-explorer/mango/combinableinstructions.py", line 234, in execute
    response = context.client.send_transaction(transaction, *self.signers)
  File "/root/mango/mango-explorer/mango/client.py", line 786, in send_transaction
    response = self.compatible_client.send_transaction(transaction, *signers, opts=proper_opts)
  File "/usr/local/lib/python3.9/dist-packages/solana/rpc/api.py", line 1185, in send_transaction
    txn_resp = self.send_raw_transaction(txn.serialize(), opts=opts)
  File "/usr/local/lib/python3.9/dist-packages/solana/rpc/api.py", line 1137, in send_raw_transaction
    resp = self._provider.make_request(*args)
  File "/root/mango/mango-explorer/mango/client.py", line 574, in make_request
    result = provider.make_request(method, *params)
  File "/root/mango/mango-explorer/mango/client.py", line 416, in make_request
    return self.__make_request(method, *params)
  File "/root/mango/mango-explorer/mango/client.py", line 516, in __make_request
    raise TransactionException(transaction, exception_message, error_code, self.name,
mango.client.TransactionException: « TransactionException in 'Mango Explorer' [sendTransaction]: -32002:: Transaction failed with: 'Transaction simulation failed: Error processing Instruction 0: custom program error: 0x19'
    Instructions:
        « CreateSpotOpenOrders: None
                Key[ 0]: 98pjRuQjK3qA6gXts96PqZT4Ze5QmnCmt3QYjhbUSPue
                Key[ 1]: 2WkWHY1u7LyXuYZSjA2SEqNiZDyhr8VMrNQ8rfVtBxfk
                Key[ 2]: Es9NnGZRk3Ew7Cb8csXvaVpLJc9Yn7M6YKUhtqodnjh6
                Key[ 3]: 9xQeWvG816bUx9EPjHmaT23yvVM2ZWbrrpZb9PusVFin
                Key[ 4]: TK4VWqe3Z2eV4ykNcMPs2NDapGYqQBqFY4e5VaASSjK
                Key[ 5]: 9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT
                Key[ 6]: 9BVcYqEQxyccuwznvxXqDkSJFavvTyheiTYk231T1A8S
                Key[ 7]: 11111111111111111111111111111111
        »
        Instruction Details:
        Key[ 0]: 98pjRuQjK3qA6gXts96PqZT4Ze5QmnCmt3QYjhbUSPue  Read-Only       
        Key[ 1]: 2WkWHY1u7LyXuYZSjA2SEqNiZDyhr8VMrNQ8rfVtBxfk  Writable        
        Key[ 2]: Es9NnGZRk3Ew7Cb8csXvaVpLJc9Yn7M6YKUhtqodnjh6  Writable  Signer
        Key[ 3]: 9xQeWvG816bUx9EPjHmaT23yvVM2ZWbrrpZb9PusVFin  Read-Only       
        Key[ 4]: TK4VWqe3Z2eV4ykNcMPs2NDapGYqQBqFY4e5VaASSjK   Writable        
        Key[ 5]: 9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT  Writable        
        Key[ 6]: 9BVcYqEQxyccuwznvxXqDkSJFavvTyheiTYk231T1A8S  Read-Only       
        Key[ 7]: 11111111111111111111111111111111              Read-Only       
        Program ID: mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68
        Data: 3c000000
        « PlaceSpotOrder2: None
                Key[ 0]: 98pjRuQjK3qA6gXts96PqZT4Ze5QmnCmt3QYjhbUSPue
                Key[ 1]: 2WkWHY1u7LyXuYZSjA2SEqNiZDyhr8VMrNQ8rfVtBxfk
                Key[ 2]: Es9NnGZRk3Ew7Cb8csXvaVpLJc9Yn7M6YKUhtqodnjh6
                Key[ 3]: EBDRoayCDDUvDgCimta45ajQeXbexv7aKqJubruqpyvu
                Key[ 4]: 9xQeWvG816bUx9EPjHmaT23yvVM2ZWbrrpZb9PusVFin
                Key[ 5]: 9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT
                Key[ 6]: 14ivtgssEBoBjuZJtSAPKYgpUK7DmnSwuPMqJoVTSgKJ
                Key[ 7]: CEQdAFKdycHugujQg9k2wbmxjcpdYZyVLfV9WerTnafJ
                Key[ 8]: AZG3tFCFtiCqEwyardENBQNpHqxgzbMw8uKeZEw2nRG5
                Key[ 9]: 5KKsLVU6TcbVDK4BS6K1DGDxnh4Q9xjYJ8XaDCG5t8ht
                Key[10]: 36c6YqAwyGKQG66XEp2dJc5JqjaBNv7sVghEtJv4c7u6
                Key[11]: 8CFo8bL8mZQK8abbFyypFMwEDd8tVJjHTTojMLgQTUSZ
                Key[12]: 7jH1uLmiB2zbHNe6juZZYjQCrvquakTwd3yMaQpeP8rR
                Key[13]: 2bqJYcA1A8gw4qJFjyE2G4akiUunpd9rP6QzfnxHqSqr
                Key[14]: AVn3JRGhifPCxjxZsU3tQuo4U4dTHizHzBDGW983tx47
                Key[15]: AMzanZxMirPCgGcBoH9kw4Jzi9LFMomyUCXbpzDeL2T8
                Key[16]: BGcwkj1WudQwUUjFk78hAjwd1uAm8trh1N4CJSa51euh
                Key[17]: 8Vw25ZackDzaJzzBBqcgcpDsCsDfRSkMGgwFQ3gbReWF
                Key[18]: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA
                Key[19]: 9BVcYqEQxyccuwznvxXqDkSJFavvTyheiTYk231T1A8S
                Key[20]: F8Vyqk3unwxkXukZFQeYyGmFfTG3CAX4v24iyrjEYBJV
                Key[21]: HiB9JtxgnA7G29EWVcPFhXbLCTaPitdVJKgD3BhH6TJj
                Key[22]: TK4VWqe3Z2eV4ykNcMPs2NDapGYqQBqFY4e5VaASSjK
        »
        Instruction Details:
        Key[ 0]: 98pjRuQjK3qA6gXts96PqZT4Ze5QmnCmt3QYjhbUSPue  Read-Only       
        Key[ 1]: 2WkWHY1u7LyXuYZSjA2SEqNiZDyhr8VMrNQ8rfVtBxfk  Writable        
        Key[ 2]: Es9NnGZRk3Ew7Cb8csXvaVpLJc9Yn7M6YKUhtqodnjh6  Writable  Signer
        Key[ 3]: EBDRoayCDDUvDgCimta45ajQeXbexv7aKqJubruqpyvu  Read-Only       
        Key[ 4]: 9xQeWvG816bUx9EPjHmaT23yvVM2ZWbrrpZb9PusVFin  Read-Only       
        Key[ 5]: 9wFFyRfZBsuAha4YcuxcXLKwMxJR43S7fPfQLusDBzvT  Writable        
        Key[ 6]: 14ivtgssEBoBjuZJtSAPKYgpUK7DmnSwuPMqJoVTSgKJ  Writable        
        Key[ 7]: CEQdAFKdycHugujQg9k2wbmxjcpdYZyVLfV9WerTnafJ  Writable        
        Key[ 8]: AZG3tFCFtiCqEwyardENBQNpHqxgzbMw8uKeZEw2nRG5  Writable        
        Key[ 9]: 5KKsLVU6TcbVDK4BS6K1DGDxnh4Q9xjYJ8XaDCG5t8ht  Writable        
        Key[10]: 36c6YqAwyGKQG66XEp2dJc5JqjaBNv7sVghEtJv4c7u6  Writable        
        Key[11]: 8CFo8bL8mZQK8abbFyypFMwEDd8tVJjHTTojMLgQTUSZ  Writable        
        Key[12]: 7jH1uLmiB2zbHNe6juZZYjQCrvquakTwd3yMaQpeP8rR  Read-Only       
        Key[13]: 2bqJYcA1A8gw4qJFjyE2G4akiUunpd9rP6QzfnxHqSqr  Writable        
        Key[14]: AVn3JRGhifPCxjxZsU3tQuo4U4dTHizHzBDGW983tx47  Writable        
        Key[15]: AMzanZxMirPCgGcBoH9kw4Jzi9LFMomyUCXbpzDeL2T8  Read-Only       
        Key[16]: BGcwkj1WudQwUUjFk78hAjwd1uAm8trh1N4CJSa51euh  Writable        
        Key[17]: 8Vw25ZackDzaJzzBBqcgcpDsCsDfRSkMGgwFQ3gbReWF  Writable        
        Key[18]: TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA   Read-Only       
        Key[19]: 9BVcYqEQxyccuwznvxXqDkSJFavvTyheiTYk231T1A8S  Read-Only       
        Key[20]: F8Vyqk3unwxkXukZFQeYyGmFfTG3CAX4v24iyrjEYBJV  Read-Only       
        Key[21]: HiB9JtxgnA7G29EWVcPFhXbLCTaPitdVJKgD3BhH6TJj  Read-Only       
        Key[22]: TK4VWqe3Z2eV4ykNcMPs2NDapGYqQBqFY4e5VaASSjK   Writable        
        Program ID: mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68
        Data: 2900000000000000905f0100000000000000000000000000000000000000000000000000000000007e6fa0b67e010000ffff
    Accounts:
        No Accounts
    Errors:
        {'InstructionError': [0, {'Custom': 25}]}
    Logs:
        Program mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68 invoke [1]
        Program log: Mango: CreateSpotOpenOrders
        Program log: Custom program error: 0x19
        Program mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68 consumed 9634 of 200000 compute units
        Program mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68 failed: custom program error: 0x19
    Response:
        {"jsonrpc":"2.0","error":{"code":-32002,"message":"Transaction simulation failed: Error processing Instruction 0: custom program error: 0x19","data":{"accounts":null,"err":{"InstructionError":[0,{"Custom":25}]},"logs":["Program mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68 invoke [1]","Program log: Mango: CreateSpotOpenOrders","Program log: Custom program error: 0x19","Program mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68 consumed 9634 of 200000 compute units","Program mv3ekLzLbnVPNxjSKvqBpU3ZeZXPQdEC3bp5MDEBG68 failed: custom program error: 0x19"]}},"id":16}
```

**Comment by OpinionatedGeek** [comment]
Hmm. I can tell you that it's having difficulty creating the spot OpenOrders address, and that the code there changed recently. (It was working when I wrote it, but maybe something else is interfering?)

In any case, as a workaround you could use the Mango UI to place an order on that market - just once, and the order doesn't have to fill. That will create the OpenOrders account for that market and your place-order commands should work (for that account+market) from then on.

**Comment by grossman8** [comment]
Thanks, creating a single order with the UI cleared up the issue.

**Comment by OpinionatedGeek** [comment]
Excellent. Should you need it, this issue has been fixed and the fix has been released in v3.3.4.

Thanks for raising this issue.
